src/api/java/baritone/api/utils/command/Command.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.utils.command;
+
+import baritone.api.BaritoneAPI;
+import baritone.api.IBaritone;
+import baritone.api.Settings;
+import baritone.api.event.listener.AbstractGameEventListener;
+import baritone.api.utils.Helper;
+import baritone.api.utils.IPlayerContext;
+import baritone.api.utils.command.execution.CommandExecution;
+import baritone.api.utils.command.helpers.arguments.ArgConsumer;
+import net.minecraft.client.Minecraft;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class Command implements Helper, AbstractGameEventListener {
+    protected IBaritone baritone = BaritoneAPI.getProvider().getPrimaryBaritone();
```

Comment:
Well what do you know, this is causing issues. You have a circular service loader issue causing the Travis failure.

BaritoneAPI.getProvider() is called, which loads the service, which instantiates BaritoneProvider, which instantiates Baritone, which instantiates the commands, which calls BaritoneAPI.getProvider..... before returning the BaritoneProvider. So BaritoneAPI.getProvider() is returning null at this point, since it's too early.

Reply:
> Well what do you know, this is causing issues.

I'm sorry what? Where did this sudden hostility come from? I believe it was your advice in the first place that led to this.

Reply:
> instantiates Baritone, which instantiates the commands

Actually chat control is instantiated in BaritoneProvider.

Nonetheless I'll look into this

Reply:
I don't believe that's true... I'm fairly sure that I've been critical all along of these helper fields being in Command instead of in an actual Helper interface, haven't I?

Reply:
Possibly. If you want I can refactor this whole thing and make commands work for more than one Baritone

Reply:
Even though I don't think that's the goal, I don't think it would be a problem. I believe that in order to solve this issue you'd need each command to have its own baritone, so if you add into the constructor of each Command that it has to take in a IBaritone and pass it to super, that would solve both issues (the provider meme and the nebulous "but what if i actually make bot system")

Reply:
There are two ways this can go:
- Each Baritone instance has its own command manager and etc.
- Each command can get called on any Baritone at any time

Issues:
- Registering commands. However, there may be utility methods for this
- Commands that keep state. pause and resume come to mind, this will have to be isolated per baritone.

I'm leaning towards the first one. Each Baritone will have its own command manager, so in the future if there's more than one, there can be a command to execute a command on another Baritone (or all of them). I believe that is the best one.

Reply:
Sounds good to me